### ISSUE TITLE
Memory Footprint of Implicit Triangulations

### ISSUE THREAD

**Issue by JonasLukasczyk** [topology-tool-kit/ttk#749]
Hi everyone,
the memory footprint of the implicit triangulation has increased significantly over the last year. Consider the following test:
1. Create a Wavelet source with a resolution of 256^3. This creates a vtkImageData object holding roughly 64 MB float data.
2. Switch to a SpreadSheet view in order to omit memory consumption caused by rendering.
3. Running now the ttkHelloWorld filter (which only requires vertex neighbors) now increases PV's memory consumption by 3GB! Note, turning the vtkImageData object into a triangle mesh with vtkTetrahedralize increases the memory consumption by 4GB. So the implicit triangulation uses almost as much memory as explicitly representing the input object in VTK.
4. An additional problem is that memory is no longer freed after the corresponding vtkDataObject was destroyed. I do not know at which point this feature broke. Maybe this is related to the newer versions of VTK/ParaView.

I'm not able to run this experiment with a 512^3 volume on my machine with 18 GB ram, which should be more than enough for processing 0.5 GB of data.

In short, the implicit triangulation data structure performs too aggressive caching. This negates the primary advantage of implicit representations: the low memory footprint. I do not even know if this caching really brings a noticeable speedup that justifies such a significant memory hit.

I suggest to solve the issue by reverting the implicit triangulation to what the name suggests and either provide the cached versions as alternative triangulation data structures (e.g., ttkImplicitTriangulationCached) or toggle caching at compile time. What are your thoughts @julien-tierny, @CharlesGueunet, @chgarth, @pierre-guillou?

Best
Jonas

**Comment by julien-tierny** [comment]
we are on it thanks.
the pr is in the works.
a triangulation manager will now centralize all triangulation parameters (periodicity, compact explicit support etc). it will enable to switch back and forth between the representations. we'll double check the memory issue.
